validate.mjs

```diff
@@ -1,25 +1,28 @@
 import { readFile } from 'fs/promises'
 import Ajv from 'ajv/dist/2019.js'
 import meta from 'ajv/dist/refs/json-schema-2020-12/index.js'
+import { glob } from 'glob'
 
 const ajv = new Ajv({ verbose: true })
 ajv.addMetaSchema(meta, 'https://json-schema.org/draft/2020-12/schema')
 
 /** @param {string} path */
 async function validate(path) {
-  const text = await readFile(path, { encoding: 'utf-8' })
-  const schema = JSON.parse(text)
-  try {
-    ajv.compile(schema)
-    return true
-  } catch (e) {
-    if (e instanceof Error) {
-      process.exitCode = 1
-      console.error(`${path}: ${e.message}`)
-    } else {
-      throw e
-    }
-  }
+  const filesInDir = await glob(path, {
+    ignore: ['test/**', 'node_modules/**'],
+  })
+
+  const schemas = await Promise.all(
+    filesInDir.map(async (file) =>
+      JSON.parse(
+        await readFile(file, {
+          encoding: 'utf-8',
+        })
+      )
+    )
+  )
+  ajv.addSchema(schemas).compile(true)
```

Comment:
What happens on error?

Can we please retain the original behavior that:

* if there are no errors, the process exits cleanly
* if there are any validation errors, they are all printed with the format `${path}: ${e.message}`, and then the process exits with status 1
* if there are any errors that are not validation errors, then they are re-thrown